identitykit/src/main/java/com/eldersoss/identitykit/ext/NetworkResponseExt.kt

```diff
@@ -0,0 +1,23 @@
+package com.eldersoss.identitykit.ext
+
+import com.eldersoss.identitykit.errors.*
+import com.eldersoss.identitykit.network.NetworkResponse
+import com.eldersoss.identitykit.oauth2.Token
+
+fun NetworkResponse.parseToken(): Token {
+
+    if (this.statusCode in 400..499) {
+
+        // try found OAuth2Error or throw general Error
+        throw OAuth2Error.getError(this)
```

Comment:
Throw this error only if it can be created. Otherwise leave it to token initialization to throw error.